src/typescript/azure-openapi-validator/rules/DescriptionTooShort.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+import { trimDescription } from './utilities';
+
+export const DescriptionTooShort: string = "DescriptionTooShort";
+const minimumDescriptionLength = 20;
+
+rules.push({
+  id: "D4002",
+  name: DescriptionTooShort,
+  severity: "error",
+  category: "DocViolation",
+  mergeState: MergeStates.composed,
+  openapiType: OpenApiTypes.doc,
+  appliesTo_JsonQuery: "$..description",
+  run: function* (doc, node, path) {
+    const msg: string = "Description length is too short.";
+    if (typeof (node) !== 'string') {
+      return;
+    }
+
+    // Skip 0-length nodes; these are picked up by another validator.
+    const description: string = trimDescription(node);
+    if (node.length > 0 && description.length < minimumDescriptionLength) {
```

Comment:
Maybe check description for null too

Reply:
Also, shouldn't you trim this?

Reply:
Trimmed in the line above, but that's done before any nil check. Will do both here.

Reply:
Actually just did a full context check here - neither a nil check or a trim are necessary. The `typeof (node)` is checked earlier for `string`, so it must be non-null.